pallets/system/src/submitted_data.rs

```diff
@@ -122,7 +122,7 @@ where
 		.collect()
 }
 
-// Supports both v1 & v2 headers
+// Supports v3 headers
 fn extract_and_inspect_v2<E>(
```

Comment:
@Leouarz I think we should eliminate the dead_code from this file as well, functions like `filter`, `extract`, `process_calls` etc. are required only when using the V1 header version.

Reply:
Done